### PR TITLE
Feature/add subgraph for dpos

### PR DESCRIPTION
**Pull request by VargaElod23**

### CODE REVIEW

**Review by Malak67 — DISMISSED** [review]
LGTM

File: packages/dpos-subgraph/schema.graphql [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  delegator: String!
+  validator: String!
+}
+
+type CommissionChange @entity(immutable: true) {
+  id: ID!
+  commission: Int!
+  registrationBlock: Int!
+  applianceBlock: Int!
+  validator: String!
```
Comment (rattrap) [inline packages/dpos-subgraph/schema.graphql:54]: Let's add the date here too.

**Review by rattrap — CHANGES_REQUESTED** [review]
There's no need to store the undelegations. These can be queried from the contract

File: packages/dpos-subgraph/src/mapping.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  const currentDelegationDataTo = getOrInitCurrentDelegation(to, delegator);
+  const currentDelegationDataFrom = getOrInitCurrentDelegation(from, delegator);
+
+  // Update delegations
+  updateOrAddDelegation(currentDelegationDataFrom, amount, false);
+  updateOrAddDelegation(currentDelegationDataTo, amount, true);
+}
+
+export function handleCommissionSet(event: CommissionSet): void {
+  const commission = event.params.commission;
```
Comment (rattrap) [inline packages/dpos-subgraph/src/mapping.ts:162]: I think we can normalise the commission here to 0-100%

**Review by rattrap — CHANGES_REQUESTED** [review]

File: packages/dpos-subgraph/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#   commissionChanges: [CommissionChange!]! @derivedFrom(field: "validator")
+#   # isEligible: Boolean
+#   # info: BasicInfo!
+# }
+
+type Delegation @entity(immutable: false) {
+  id: ID!
+  amount: BigInt!
+  delegator: String!
+  validator: String!
```
[the remark this answers is not in the document]
Comment (rattrap) [reply packages/dpos-subgraph/schema.graphql:38]: Yes. That's fine.

File: packages/dpos-subgraph/src/mapping.ts (hunk 2 above)
Comment (rattrap) [reply packages/dpos-subgraph/src/mapping.ts:162]: I understand. Let's keep it as is then.

File: packages/dpos-subgraph/schema.graphql (hunk 3 above)
[the remark this answers is not in the document]
Comment (VargaElod23) [reply packages/dpos-subgraph/schema.graphql:38]: Cool, I'll be adding it in that manner then

**Review by Malak67 — APPROVED** [review]

**Pull request merged**